imageengine/class-imagecdn.php

```diff
@@ -46,6 +47,24 @@ public function __construct() {
 		add_filter( 'plugin_action_links_' . IMAGE_CDN_BASE, array( self::class, 'add_action_link' ) );
 	}
 
+	/**
+	 * Add http headers for Client Hints, Feature Policy and Preconnect Resource Hint.
+	 */
+	public static function add_headers() {
+		// Add client hints.
+        header( 'Accept-CH: viewport-width, width, device-memory, dpr, rtt, downlink, ect' );
+
+		// Add resource hints and feature policy.
+		$options = self::get_options();
+		$host    = wp_parse_url( $options['url'], PHP_URL_HOST );
+		if ( ! empty( $host ) ) {
+    		$protocol = (is_ssl()) ? "https://" : "http://";
+        header( 'Link: <'.$protocol.$host.'>; rel=preconnect' );
+        header( 'Feature-Policy: ch-viewport-width '.$protocol.$host.'; ch-width '.$protocol.$host.'; ch device-memory '.$protocol.$host.'; ch-dpr '.$protocol.$host.'; ch-rtt '.$protocol.$host.'; ch-downlink '.$protocol.$host.'; ch-ect '.$protocol.$host.';' );
```

Comment:
I think this would be a lot cleaner using string interpolation like this:
```php
$url = $protocol.$host;
header( "Link: <$url>; rel=preconnect" );
header( "Feature-Policy: ch-viewport-width $url; ch-width $url; ch device-memory $url; ch-dpr $url; ch-rtt $url; ch-downlink $url; ch-ect $url;" );
```